tests/unit/s2n_pq_kem_kat_kyber_r3_test.c

```diff
@@ -36,7 +36,7 @@ static const struct s2n_kem_kat_test_vector test_vectors[] = {
 
 int main() {
     BEGIN_TEST();
-    if (!s2n_pq_is_enabled()) {
+    if (!s2n_pq_is_enabled() || s2n_aws_lc_supports_kyber_512() ) {
```

Comment:
Hmm could you elaborate on this?

Reply:
I wrote about it a bit in the call outs and also talked to a few others about it offline. To add to that I don't think it's possible (or at least easy) to seed the RNG of AWS-LC using the exposed interfaces and that makes implementing this test impossible.

> The s2n_pq_kem_kat_kyber_r3_test test case is now skipped when S2N is built using [AWS-LC](https://github.com/awslabs/aws-lc/blob/ed738e7201d22db96ea8bd14e46b803ef0566c18/crypto/evp_extra/pq_kem_kat_tests_kyber512.cc)'s Kyber implementation. A redundant implementation of this test exists in AWS-LC. Fixing this for S2N will require seeding AWS-LC's RNG properly to force determinism in the Kyber key generation algorithm. This level of depth is not appropriate in the case where S2N is not using its own Kyber implementation. Other S2N unit tests validate that AWS-LC's Kyber implementation can generate a key and use it to perform encapsulate/decapsulate operations.

Reply:
That's right, there is no exported API that can mutate/reconfigure the AWS-LC randomness state or operation.

Reply:
Can you put the explanation in a comment?